src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -214,6 +214,12 @@ private[snowflake] class JDBCWrapper {
   private def isQuoted(name: String): Boolean = {
     name.startsWith("\"") && name.endsWith("\"") 
   }
+  private def quotedName(name: String): String = {
+      if (name.matches("[_A-Z]([_0-9A-Z])*"))
+        "\"" + name.toUpperCase + "\""
+      else
+        name
```

Comment:
I think we still want to add quotes here...we always quote identifiers now, right? The `if` only checks whether we promote to upper-case.